src/ContosoCrafts.Web.Server/Controllers/CheckoutController.cs

```diff
@@ -42,7 +42,7 @@ public class CheckoutController : ControllerBase
         }
 
         [HttpPost]
-        public async Task<ActionResult> CheckoutOrder(IEnumerable<CartItem> items, [FromServices] IServiceProvider sp)
+        public async Task<ActionResult> CheckoutOrder([FromBody]IEnumerable<CartItem> items, [FromServices] IServiceProvider sp)
```

Comment:
I needed to add `[FromBody]` to get items to parse as expected. Did I miss something there? :D